homeassistant/components/rest/data.py

```diff
@@ -35,20 +38,17 @@ def __init__(
         self.data = None
         self.headers = None
 
-    async def async_remove(self):
-        """Destroy the http session on destroy."""
-        if self._async_client:
-            await self._async_client.aclose()
-
     def set_url(self, url):
         """Set url."""
         self._resource = url
 
+    @callback
+    def async_setup(self, hass):
```

Comment:
Any reason we would do it during setup instead of having this be still called within `async_update` ? We don't even have to store it as an instance variable anymore either.

Reply:
I wanted to avoid passing `hass` to `async_update`, but I guess it really doesn't matter to add that.

Reply:
Removed async_setup